app/controllers/ontologies_controller.rb

```diff
@@ -168,6 +168,9 @@ def mappings
   end
 
   def new
+    if $READ_ONLY_PORTAL && !session[:user]&.admin?
```

Comment:
create a helper for thins in application_helper.rb, something like
```ruby
def read_only_enabled?
    $READ_ONLY_PORTAL && !session[:user]&.admin?
end 
```

and you will call it like this
```ruby
 redirect_to "/login?redirect=#{new_ontology_path}" if read_only_enabled?
```